pkg/custompluginmonitor/plugin/plugin.go

```diff
@@ -127,14 +133,50 @@ func (p *Plugin) run(rule cpmtypes.CustomRule) (exitStatus cpmtypes.Status, outp
 	defer cancel()
 
 	cmd := exec.CommandContext(ctx, rule.Path, rule.Args...)
-	stdout, err := cmd.Output()
+
+	stdoutPipe, err := cmd.StdoutPipe()
+	if err != nil {
+		glog.Errorf("Error creating stdout pipe for plugin %q: error - %v", rule.Path, err)
+		return cpmtypes.Unknown, "Error creating stdout pipe for plugin. Please check the error log"
+	}
+	stderrPipe, err := cmd.StderrPipe()
+	if err != nil {
+		glog.Errorf("Error creating stderr pipe for plugin %q: error - %v", rule.Path, err)
+		return cpmtypes.Unknown, "Error creating stderr pipe for plugin. Please check the error log"
+	}
+	stdoutReader := io.LimitReader(stdoutPipe, maxCustomPluginBufferBytes)
+	stderrReader := io.LimitReader(stderrPipe, maxCustomPluginBufferBytes)
+
+	if err := cmd.Start(); err != nil {
+		glog.Errorf("Error in starting plugin %q: error - %v", rule.Path, err)
+		return cpmtypes.Unknown, "Error in starting plugin. Please check the error log"
+	}
+
```

Comment:
Should we read stdout and stderr at the same time? Or else, the stderr side may potentially block the plugin process when the buffer is full?

And we should probably continue reading the output and discard them after reaching the limit, or else the plugin process may be blocked as well.

Reply:
Makes sense. Reading from stdout and stderr at the same time.
Also added discard once we have read the maxBytes.